show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,184 @@
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.esri.arcgismaps.sample.showgeodesicpathbetweentwopoints.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // set up data binding for the mapView
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // shows the distance information as a textview
+    private val infoTextView by lazy {
+        activityMainBinding.infoTextView
+    }
+
+    // a blue marker symbol for the location points
+    private val locationMarkerSymbol by lazy {
+        SimpleMarkerSymbol(
+            SimpleMarkerSymbolStyle.Circle,
+            Color.red,
+            10f
+        )
+    }
+
+    // the marker graphic for the starting location
+    private val startingLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+            geometry = startingPoint
+        }
+    }
+
+    // marker graphic for the destination
+    private val endLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+        }
+    }
+
+    // the geodesic path represented as line graphic
+    private val geodesicPathGraphic by lazy {
+        val lineSymbol = SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.red, 5f)
+        Graphic().apply {
+            symbol = lineSymbol
+        }
+    }
+
+    // the unit of distance measurement in kilometers
+    private val unitsOfMeasurement = LinearUnit(LinearUnitId.Kilometers)
+
+
+    // starting location for the distance calculation
+    private val startingPoint = Point(-73.7781, 40.6413, SpatialReference.wgs84())
+
+
+    // creates a graphic overlay to draw all graphics
+    private val graphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISImageryStandard)
```

Comment:
```suggestion
        val map = ArcGISMap(BasemapStyle.ArcGISImageryStandard).apply {
            initialViewpoint = Viewpoint(Point(34.77, -10.24), 20e7)
        }
```

Just a suggestion, I feel setting the viewpoint to fill the screen to the map would be a nice addition to the sample.